hal_st/middlewares/ble_middleware/GattClientSt.cpp

```diff
@@ -38,6 +38,66 @@ namespace hal
         aci_gatt_disc_all_char_desc(connectionHandle, service.Handle(), service.EndHandle());
     }
 
+    void GattClientSt::Read(const services::GattClientCharacteristicOperationsObserver& characteristic, const infra::Function<void(const infra::ConstByteRange&)>& onResponse) const
+    {
+        this->onResponse = onResponse;
+
+        aci_gatt_read_char_value(connectionHandle, characteristic.CharacteristicValueHandle());
+    }
+
+    void GattClientSt::Write(const services::GattClientCharacteristicOperationsObserver& characteristic, infra::ConstByteRange data, const infra::Function<void()>& onDone) const
+    {
+        this->onDone = onDone;
+
+        aci_gatt_write_char_value(connectionHandle, characteristic.CharacteristicValueHandle(), data.size(), data.cbegin());
+    }
+
+    void GattClientSt::WriteWithoutResponse(const services::GattClientCharacteristicOperationsObserver& characteristic, infra::ConstByteRange data) const
+    {
+        aci_gatt_write_without_resp(connectionHandle, characteristic.CharacteristicValueHandle(), data.size(), data.cbegin());
+    }
+
+    void GattClientSt::EnableNotification(const services::GattClientCharacteristicOperationsObserver& characteristic, const infra::Function<void()>& onDone) const
+    {
+        this->onDone = onDone;
+
+        auto characteristicValue = services::GattDescriptor::ClientCharacteristicConfiguration::CharacteristicValue::enableNotification;
+
+        if ((characteristic.CharacteristicProperties() & services::GattCharacteristic::PropertyFlags::notify) == services::GattCharacteristic::PropertyFlags::notify)
+            aci_gatt_write_char_desc(connectionHandle, characteristic.CharacteristicValueHandle() + 1, sizeof(characteristicValue), reinterpret_cast<uint8_t*>(&characteristicValue));
+    }
+
+    void GattClientSt::DisableNotification(const services::GattClientCharacteristicOperationsObserver& characteristic, const infra::Function<void()>& onDone) const
+    {
+        this->onDone = onDone;
+
+        auto characteristicValue = services::GattDescriptor::ClientCharacteristicConfiguration::CharacteristicValue::disable;
+
+        if ((characteristic.CharacteristicProperties() & services::GattCharacteristic::PropertyFlags::notify) == services::GattCharacteristic::PropertyFlags::notify)
+            aci_gatt_write_char_desc(connectionHandle, characteristic.CharacteristicValueHandle() + 1, sizeof(characteristicValue), reinterpret_cast<uint8_t*>(&characteristicValue));
```

Comment:
I see this calculation at a lot of places. 
Previously we made it to         
```c++

constexpr uint8_t offsetCCCD = 1;
```
How can we have it in a central place such as GattChar?

Reply:
We definitely could, but also requires changes on emil =)

Reply:
Aligned via call: planned to be updated on the coming sprints.